### ISSUE TITLE
Generating graphs with dot does not handle long paths well

### ISSUE THREAD

**Issue by rancier** [copperspice/doxypress#25]
Hi all!

I've attahced a zip, inside two folders and two Doxy JSONS. If you put the two folders to the C:\ root directory and run the doxy_sample7.json then everything will be everything fine.

But if you run doxy_sample8.jsoon you will get the following error message:
Error: Unable to run 'C:/Program Files (x86)/Graphviz2.38/bin/dot.exe', exit code = 2
Arguments = '"./inline_dotgraph_1.dot" -Tsvg -o "C:/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx12345678/doc/dot_inline_dotgraph_1.svg"'

The two JSONs and the two directories differs only in directory name (...8 is 1 character longer). 
If you run the dot.exe with the full path it will generate the files correctly, so I think the bug is in the doxypress.
[doxy_sample.zip](https://github.com/copperspice/doxypress/files/2936968/doxy_sample.zip)

**Comment by rancier** [comment]
Versions:
c:\Program Files\DoxyPress>doxypress.exe --version

DoxyPress Version: 1.3.3
email: info@copperspice.com
---------------------
c:\Program Files (x86)\Graphviz2.38\bin>dot -V
dot - graphviz version 2.38.0 (20140413.2041)

**Comment by agserm** [comment]
Thank you for the report. I looked at an equivalent project on a Linux machine and it seemed to work fine. It seems like is most likely a platform specific issue which only occurs on Windows. Can you let us know what operating system version you are running this on? Also, can you confirm that the graphviz package you are using is the official one from graphviz.org? We will work on reproducing this in our test environment shortly.

**Comment by rancier** [comment]
Operating system: Windows 10 Enterprise, Version 1709, Build: 16299.904
Graphviz: Graphviz 2.38 (https://graphviz.gitlab.io/_pages/Download/windows/graphviz-2.38.msi)

**Comment by rancier** [comment]
I've examined the behaviour of doxypress.exe with Process Monitor (like strace in Linux). In the good case the parameters of dot.exe:

"C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" "./inline_dotgraph_1.dot" -Tcmapx -o "C:/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx1234567/doc/dot_inline_dotgraph_1.map"

"C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" "./inline_dotgraph_1.dot" -Tsvg -o "C:/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx1234567/doc/dot_inline_dotgraph_1.svg"

In the bad case:
"C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" ۏ
"C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" 꼠ۏ

**Comment by Sp3EdeR** [comment]
Upon investigation, I could reproduce this issue using the supplied samples, but it is not in connection to the path length. It is completely random and it is roughly reproducible 30% of the time. I will look into what is happening here...

**Comment by Sp3EdeR** [comment]
This issue turns out to be quite simple. In `portable.cpp`, the `SHELLEXECUTEINFOW sInfo = {` structure is initialized with a fairly long init list. Inside that init list the following two strings are passed through:
```
command.toStdWString().c_str(),     /* file to execute */
args.toStdWString().c_str(),        /* argument list */
```
Please move the wstring objects out to before the initialization as:
```
auto commandW = command.toStdWString();
auto argsW = args.toStdWString();
```
And replace the initialization list contents with:
```
commandW.c_str();
argsW.c_str();
```

The issue is that the temporary `wstring` objects are destroyed before their contents are used in the subsequent `ShellExecuteExW` call.

**Comment by agserm** [comment]
Closing this ticket since the fix you provided has been merged in cd842ce. Thank you for your contribution, great debugging!
